### PR TITLE
Stablize tests using random

### PR DESCRIPTION
**Pull request by shanjiaz**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
This test might be a little flaky. Adding seed to stablize it. 

## Description
Added `torch.manual_seed(42)`

## Tests
Tested locally. 

I have filled in:

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan/results, such as providing test command and pasting the results.
- [ ] (Optional) The necessary documentation update.
- [x] I (a human) have written or reviewed the code in this pr to the best of my ability.

### CODE REVIEW

File: tests/unit/models/test_dflash_metrics.py [hunk 1]
```diff
@@ -154,6 +154,7 @@ def test_loss_mask_zeros_out_positions(self):
     def test_different_gamma(self):
         """Different gamma values should produce different losses."""
         B, T, V = 1, 16, 10
+        torch.manual_seed(42)
```
Comment (rahul-tuli) [inline tests/unit/models/test_dflash_metrics.py]: instead of repeating this line, could we make this into an autouse fixture? will help stabilize other tests as well, something like:

```python
import pytest
import numpy as np
import torch

@pytest.fixture(autouse=True)
def seed_everything(seed: int = 42):
    np.random.seed(seed)
    torch.manual_seed(seed)
    yield
```

**Review by rahul-tuli — DISMISSED** [review]

File: tests/unit/models/test_dflash_metrics.py (hunk 1 above)
Comment (shanjiaz) [reply tests/unit/models/test_dflash_metrics.py]: sure

**Review by fynnsu — APPROVED** [review]
